Modules/Filtering/ImageFeature/include/itkFastBilateralImageFilter.h

```diff
@@ -0,0 +1,204 @@
+
+  /** Pixel types. */
+  using OutputPixelType = typename TOutputImage::PixelType;
+  using InputPixelType = typename TInputImage::PixelType;
+
+  /** Typedef for an array of doubles that specifies the DomainSigma
+   *  in each spacial dimension. */
+  using DomainSigmaArrayType = FixedArray<double, itkGetStaticConstMacro(ImageDimension)>;
+
+  /** Standard get/set macros for filter parameters.
+   *  DomainSigma is specified in the same units as the Image spacing.
+   *  RangeSigma is specified in the units of intensity. */
+  itkGetConstMacro(DomainSigma, const DomainSigmaArrayType);
+  itkSetMacro(DomainSigma, DomainSigmaArrayType);
+  itkGetConstMacro(RangeSigma, double);
+  itkSetMacro(RangeSigma, double);
+
+  /** Convenience set method for setting all domain standard deviations to the
+   *  same value. */
+  void
+  SetDomainSigma(const double v)
+  {
+    m_DomainSigma.Fill(v);
+  }
+
+protected:
+  /** Default Constructor. Default value for DomainSigma is 4. Default
+   *  value for RangeSigma is 50. These values were chosen match those of the
+   *  BilateralImageFilter */
+  FastBilateralImageFilter()
+  {
+    m_DomainSigma.Fill(4.0);
+    m_RangeSigma = 50.0;
+  }
+
+  virtual ~FastBilateralImageFilter() {}
+
+  /*
+   * The FastBilateralImageFilter needs a larger input requested
+   * region than the size of the output requested region. Like
+   * the BilateralImageFilter, the FastBilateralImageFilter needs
+   * an amount of padding in each dimension based on the domain sigma.
+   */
+  void
+  GenerateInputRequestedRegion() override;
+
+  /** Standard pipeline method */
+  void
+  GenerateData() override;
+
+  /** Method to print member variables to an output stream */
+  void
+  PrintSelf(std::ostream & os, Indent indent) const override;
+
+  /** The type of image to use as the higher dimensional grid.
+   * The blurring is performed on this image type. */
+  using GridType = typename itk::Image<float, itkGetStaticConstMacro(ImageDimension) + 1>;
+
+  /** Grid types */
+  using GridPixelType = typename GridType::PixelType;
+  using GridIndexType = typename GridType::IndexType;
+  using GridSizeType = typename GridType::SizeType;
+  using GridSizeValueType = typename Size<itkGetStaticConstMacro(ImageDimension) + 1>::SizeValueType;
+  using GridRegionType = typename GridType::RegionType;
+
+  /** Grid image iterator type. */
+  using GridImageIteratorType = ImageRegionIterator<GridType>;
+  using GridImageConstIteratorType = ImageRegionConstIterator<GridType>;
+
+  /** The type of blurring to use on the grid. */
+  using BlurType = DiscreteGaussianImageFilter<GridType, GridType>;
+
+  /** The type of interpolation done to calculate output pixels. */
+  using InterpolatorType = LinearInterpolateImageFunction<GridType, float>;
+  using InterpolatedIndexType = typename InterpolatorType::ContinuousIndexType;
+
+  double               m_RangeSigma;
+  DomainSigmaArrayType m_DomainSigma;
```

Comment:
Nowadays we would use data member initializers:

    double               m_RangeSigma{ 50.0 };
    DomainSigmaArrayType m_DomainSigma{ DomainSigmaArrayType::Filled(40.0) };